executor/aggfuncs/func_group_concat.go

```diff
@@ -82,8 +86,17 @@ func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 		}
 	}
 	p.buffer.Truncate(p.buffer.Len() - len(e.sep))
-	// TODO: if total length is greater than global var group_concat_max_len, truncate it.
-	// issue: #7034
+	if e.maxLen > 0 && uint64(p.buffer.Len()) > e.maxLen {
+		i := mathutil.MaxInt
+		if uint64(i) > e.maxLen {
+			i = int(e.maxLen)
+		}
+		p.buffer.Truncate(i)
+		if !e.truncated {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(expression.ErrCutValueGroupConcat)
+		}
+		e.truncated = true
```

Comment:
1. why not set this in line 163?
2. Can these duplicated codes be extracted as a function?

Reply:
1. forgotten.... :(
2. ok